optaplanner-core/src/main/java/org/optaplanner/core/config/solver/SolverConfig.java

```diff
@@ -647,6 +648,7 @@ public SolverConfig inherit(SolverConfig inheritedConfig) {
                 inheritedConfig.getScoreDirectorFactoryConfig());
         terminationConfig = ConfigUtils.inheritConfig(terminationConfig, inheritedConfig.getTerminationConfig());
         phaseConfigList = ConfigUtils.inheritMergeableListConfig(phaseConfigList, inheritedConfig.getPhaseConfigList());
+        metricConfig = ConfigUtils.inheritConfig(metricConfig, inheritedConfig.getMetricConfig());
```

Comment:
How do we make sure that this is not exposed to the users?
(Geoffrey wants to keep in control of the solver config. But he will be gone for 3 weeks. I'd like to merge this earlier, but I won't be able to, if it impacts public API.)

Reply:
Currently, it XML transitive (i.e. can be accessed via Java API, but cannot via XML API. It need to be in the config so it can be read by the SolverFactory (which uses it to set up metrics)

Reply:
The problem with it being in the Java API is that it is a public API that needs to be kept stable.
Therefore I can not merge this PR until discussed with and approved by Geoffrey.